hardware-api.md

```diff
@@ -95,6 +96,83 @@ pin.analogRead(function(error, value) {
 });
 ```
 
+### Interrupt Pins
+
+Interrupts allow us to register events based on state changes in a pin.
+
+Take a look at the following circuit. We've connected Port A pin2 to ground
+via a button. Before we press the button, circuit is open and the voltage at
+pin2 will be high. When the button is pressed we'll complete the circuit
+and the voltage at pin2 will fall.
+
+![Interrupt Circuit](images/interrupt-circuit.png)
+
+Let's turn on the green LED when the button is pressed. Here's one way of doing
+it.
+
+```js
+var tessel = require('tessel'); // Import tessel
+
+// Initialize our pins
+var green = tessel.led[2];
+var pin2 = tessel.port.A.pin[2];
+
+// Turn off the green LED
+green.off();
+
+// Every 10 milliseconds check to see if the button was pressed
+setInterval(function() {
+  pin2.read(function(error, value) {
+
+    // button was pressed
+    if (value === 0) {
+      green.on();
+    }
+  });
+}, 10);
+
+```
+
+This is pretty inefficient. We're spending a lot of time (every 10 millseconds)
+manually checking to see if the button was pressed. Let's just tell the
+processor to run some code when the voltage drops.
+
+```js
+var tessel = require('tessel'); // Import tessel
+
+// Initialize our pins
+var green = tessel.led[2];
+var pin2 = tessel.port.A.pin[2];
+
+// Turn off the green LED
+green.off();
+
+// Register an event. When the voltage on pin2 falls, turn on the green LED.
+pin2.on('fall', function() {
+  green.on();
+});
+
+```
+
+We tell the processor to invoke our callback function when pin2 falls (button
+has been pressed). We don't have to continuously poll to see if the voltage
+has fallen. We can just register our listener and move on.
+
+#### Available interrupt events
+
+event  | description
+-------|------------
+rise   | pin voltage rises
+fall   | pin voltage falls
+change | pin voltage changes
```

Comment:
Change also passes a `value` parameter to the callback function, i.e. `pin2.on('change', function (value) {})`. So if there is a sensor attached to the interrupt, the value of the sensor will be read on `change` and passed to the callback.